src/graphql/node/control.rs

```diff
@@ -74,14 +91,924 @@ impl NodeControlMutation {
             )
         }
     }
+
+    #[graphql(guard = "RoleGuard::new(Role::SystemAdministrator)
+        .or(RoleGuard::new(Role::SecurityAdministrator))")]
+    async fn apply_node(&self, ctx: &Context<'_>, id: ID) -> Result<ApplyResult> {
+        let i = id.as_str().parse::<u32>().map_err(|_| "invalid ID")?;
+        let agents = ctx.data::<BoxedAgentManager>()?;
+
+        let node: Node = {
+            let store = crate::graphql::get_store(ctx).await?;
+            let node_map = store.node_map();
+            node_map
+                .get_by_id(i)?
+                .ok_or_else(|| async_graphql::Error::new(format!("Node with ID {i} not found",)))?
+        };
+
+        if node.name_draft.is_none() && node.settings_draft.is_none() {
+            return Err("There is nothing to apply.".into());
+        }
+
+        let review_config_setted = match &node.settings_draft {
+            Some(settings_draft) if settings_draft.review => {
+                // TODO: Trigger `set_review_config()`; Configs that can be set
+                // for REview are `revew_port` and `review_web_port`.
+                // `set_review_config` is expected to exist in REview.
+                // `set_review_config()` will write the update to a temporary
+                // file. Until the `set_review_config()` is implemented, the
+                // `review_config_setted` is temporarily fixed to true, but
+                // after implementation, it should be set according to the
+                // return value of `set_review_config()`.
+                true
+            }
+            _ => false,
+        };
+        let config_setted_modules = send_set_config_requests(agents, &node).await;
+        let success_modules = if let Ok(mut config_setted_modules) = config_setted_modules {
+            if review_config_setted {
+                config_setted_modules.push(ModuleName::Review);
+            }
+            update_node_data(ctx, i, &node, &config_setted_modules).await?;
+            if let (true, customer_id) = should_broadcast_customer_change(&node) {
+                broadcast_customer_change(customer_id, ctx).await?;
```

Comment:
Is `broadcast` a proper term? It might mean letting several receivers know about it, but since we have one receiver in this context I think we need to choose another term.

Reply:
실제로 agentmanager를 통해 리시버들에게 braodcast를 하는 기능을 하고 있으므로 broadcast 는 괜찮은 선택인 것 같긴 합니다.

Reply:
Thank you for enlightening me.